crates/fuel-core/src/schema/block.rs

```diff
@@ -83,6 +86,16 @@ pub struct PoAConsensus {
     signature: Signature,
 }
 
+#[derive(ThisError, Debug)]
+pub enum BlockError {
```

Comment:
Hmm, I'm not sure if we really want to have a custom error here. Since on the client side, it still will be a string. Maybe `anyhow::Error` will be enough to not complicate things.

Reply:
My issue is primarily that we don't own the `async_graphql::Error` and we are exposing it. On second look though, these are all non-`pub` functions, so I don't know that it actually matters.

In fact, I'm not sure who uses these functions at all. My IDE doesn't seem to know.

Reply:
Oh.... It's an `#[Object]`, that's right. I'll just put things back then. I get what's going on.